rolling-shutter/mocksequencer/shutter.go

```diff
@@ -36,24 +74,41 @@ func (s *ShutterService) SubmitBatch(transaction string) (string, error) {
 	if err != nil {
 		return "", errors.Wrap(err, "can't unmarshal incoming bytes to transaction")
 	}
-	// TODO process the BatchTx
-	// - validate the batchtx
-	// - pretty log the batch tx
-	// - decrypt and pretty log all transactions
 
-	log.Info().Msg("received batch-tx")
-	if s.processor.batchIndex != tx.BatchIndex()-1 {
-		return "", errors.New("incorrect batch-index for next batch")
+	err = s.validateBatch(&tx)
+	txStr, _ := tx.MarshalJSON()
+	if err != nil {
+		log.Ctx(ctx).Error().Err(err).RawJSON("transaction", txStr).Msg("received invalid batch transaction")
+		return "", errors.Wrap(err, "batch-tx invalid")
 	}
+	// TODO query current L1BlockNumber from L1 eth node
+	// TODO check that tx's L1BlockNumber is within (reasonable bounds) a range close to
+	//      our L1BlockNumber
+
+	// for now, just set the current view on the L1 chain to the
+	// claimed collator's view
+	currentL1BlockNumber := tx.L1BlockNumber()
+	eonKey, err := s.processor.eonKeys.Find(currentL1BlockNumber)
+	if err != nil {
+		err = errors.Wrap(err, "no eon key found for batch transaction")
+		log.Ctx(ctx).Error().Err(err).Msg("error while retrieving eon key")
+		return "", err
+	}
+
 	for _, shutterTx := range tx.Transactions() {
-		err := s.processor.processEncryptedTx(shutterTx)
+		err := s.processor.processEncryptedTx(ctx, tx.BatchIndex(), tx.L1BlockNumber(), shutterTx, tx.DecryptionKey(), eonKey)
 		if err != nil {
-			log.Error().Err(err).Msg("tx not processable, dropping transaction")
+			log.Error().Err(err).Msg("transaction invalid")
```

Comment:
Invalid transactions should result in the batch being invalid as well (at least for everything that the collator can control, i.e. all plaintext fields -- but we're not checking the decrypted part yet anyways).